sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -1505,3 +1525,34 @@ func copyOutput(cmd *exec.Cmd) error {
 	}()
 	return nil
 }
+
+// oomSniffer is a scanner that detects OOM errors in the output of a nodejs process.
+type oomSniffer struct {
+	detected bool
+}
+
+func (o *oomSniffer) Detected() bool {
+	return o.detected
+}
+
+func (o *oomSniffer) Message() string {
+	return "Detected a possible out of memory error. Consider increasing the memory available to the nodejs process " +
+		"by setting the `nodeargs` runtime option to `nodeargs: --max-old-space-size=<size>` where `<size>` is the " +
```

Comment:
Should we mention `Pulumi.yaml`? It might not be immediately obvious what the `nodeargs` runtime option is or where to set that.

Reply:
Yeah, I'll add a link to https://www.pulumi.com/docs/concepts/projects/project-file/#runtime-options as well. The message is already really long, so I think making it longer is fine.

Maybe in the future we could add an url shortener for these links.